spec/fastsync/vdd/fastsyncL2.tla

```diff
@@ -274,36 +312,37 @@ NumOfPendingRequests(bPool, peer) ==
         {h \in Heights: 
             /\ h >= bPool.height 
             /\ bPool.pendingBlocks[h] = peer
-            /\ bPool.blockStore[h] = NilBlock
         }
     IN 
     Cardinality(peerPendingRequests)
 
-\* Returns peer that can serve block for given height. 
-\* Returns NilPeer in case there are no such peer.
+(* Returns peer that can serve block for a given height. 
+   Returns NilPeer in case there are no such peer.  *)
 FindPeerToServe(bPool, h) == 
     LET peersThatCanServe == { p \in bPool.peerIds: 
                 /\ bPool.peerHeights[p] >= h 
-                /\ NumOfPendingRequests(bPool, p) <= PEER_MAX_REQUESTS } IN
+                /\ NumOfPendingRequests(bPool, p) < PEER_MAX_REQUESTS } IN
     
     LET pendingBlocks == 
         {i \in Heights: 
             /\ i >= bPool.height 
-            /\ bPool.pendingBlocks[i] /= NilPeer
-            /\ bPool.blockStore[i] = NilBlock
+            /\ \/ bPool.pendingBlocks[i] /= NilPeer
+               \/ bPool.blockStore[i] /= NilBlock
         } IN
     
     IF \/ peersThatCanServe = {} 
-       \/ Cardinality(pendingBlocks) > TARGET_PENDING 
+       \/ Cardinality(pendingBlocks) >= TARGET_PENDING 
     THEN NilPeer 
-    ELSE CHOOSE p \in peersThatCanServe: \A q \in peersThatCanServe:
+    \* pick a peer that can serve request for height h that has minimum number of pending requests
+    ELSE CHOOSE p \in peersThatCanServe: \A q \in peersThatCanServe:       
             /\ NumOfPendingRequests(bPool, p) <= NumOfPendingRequests(bPool, q) 
             /\ p < q
     
```

Comment:
This is the strategy implemented in the code. I though it is important as demonstrating some issues (removing correct peers) require fixing this strategy; otherwise it looks too abstract.